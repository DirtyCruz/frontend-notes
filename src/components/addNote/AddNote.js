import React from 'react';
import PropTypes from 'prop-types';

function AddNote({ title, handleOnChange, body, handleClick }) {
  return (
    <>
      <input type='text' name='title' value={title} onChange={handleOnChange} />
      <input type='text' name='body' value={body} onChange={handleOnChange} />
      <button onClick={handleClick}>Create Note</button>
    </>
  );
}

AddNote.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  handleOnChange: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default AddNote;
