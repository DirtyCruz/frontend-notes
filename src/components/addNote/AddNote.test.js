import React from 'react';
import { shallow } from 'enzyme';
import AddNote from './AddNote';

describe('AddNote component', () => {
  it('renders AddNote', () => {
    const wrapper = shallow(<AddNote title='title' body='body' handleOnChange={() => { }} handleClick={() => { }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
