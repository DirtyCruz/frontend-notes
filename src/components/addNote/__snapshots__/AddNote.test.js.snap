// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote component renders AddNote 1`] = `
<Fragment>
  <input
    name="title"
    onChange={[Function]}
    type="text"
    value="title"
  />
  <input
    name="body"
    onChange={[Function]}
    type="text"
    value="body"
  />
  <button
    onClick={[Function]}
  >
    Create Note
  </button>
</Fragment>
`;
